controllers/configurationpolicy_controller.go

```diff
@@ -2711,10 +2741,58 @@ func (r *ConfigurationPolicyReconciler) checkAndUpdateResource(
 
 		updateSucceeded = true
 	} else {
+		if throwSpecViolation && recordDiff != policyv1.RecordDiffNone {
+			// The spec didn't require a change but throwSpecViolation indicates the status didn't match. Handle
+			// this diff for this case.
+			mergedObjCopy := obj.existingObj.DeepCopy()
+			removeFieldsForComparison(mergedObjCopy)
+
+			// The provided isInform value is always true because the status checking can only be inform.
+			diff = handleDiff(log, recordDiff, true, existingObjectCopy, mergedObjCopy)
+		}
+
 		r.setEvaluatedObject(obj.policy, obj.existingObj, !throwSpecViolation)
 	}
 
-	return throwSpecViolation, "", updateNeeded, updateSucceeded
+	return throwSpecViolation, "", diff, updateNeeded, updateSucceeded
+}
+
+// handleDiff will generate the diff and then log it or return it based on the input recordDiff value. If recordDiff
+// is set to None or is set to InStatus with enforce, no diff is generated. When recordDiff is set to Sensored, a
+// message indicating so is returned.
+func handleDiff(
+	log logr.Logger,
+	recordDiff policyv1.RecordDiff,
+	isInform bool,
+	existingObject *unstructured.Unstructured,
+	mergedObject *unstructured.Unstructured,
+) string {
+	if recordDiff == policyv1.RecordDiffNone {
+		return ""
+	}
+
+	if !isInform && recordDiff == policyv1.RecordDiffInStatus {
+		return ""
+	}
+
+	computedDiff, err := generateDiff(existingObject, mergedObject)
+	if err != nil {
+		log.Info("Failed to generate the diff: " + err.Error())
+
+		return ""
+	}
+
+	switch recordDiff { //nolint:exhaustive
+	case policyv1.RecordDiffLog:
+		log.Info("Logging the diff:\n" + computedDiff)
+	case policyv1.RecordDiffInStatus:
+		return computedDiff
+	case policyv1.RecordDiffSensored:
+		return `# This diff may contain sensitive data. The "recordDiff" field must be set to "InStatus" ` +
```

Comment:
Why have `#` at the start?

Reply:
I was on the fence about this but the thought was that `# ` would make it a valid comment when syntax highlighting the output.

Reply:
I don't know if it will... but regardless I don't think it's a problem, and it could help the user identify these situations as 'comments' and not real diffs, so I think that's good.